R/as-data-frame.R

```diff
@@ -50,26 +50,72 @@ NULL
 #' @rdname dataset-to-R
 #' @export
 as.data.frame.CrunchDataset <- function (x, row.names = NULL, optional = FALSE,
-                                        force=FALSE, ...) {
-    out <- CrunchDataFrame(x)
-    if (force) {
-        out <- as.data.frame(out)
-    }
-    return(out)
+                                         force=FALSE, ...) {
+  out <- CrunchDataFrame(x)
+  if (force) {
+    out <- as.data.frame(out)
+  }
+  return(out)
 }
 
 #' @rdname dataset-to-R
 #' @export
 as.data.frame.CrunchDataFrame <- function (x, row.names = NULL, optional = FALSE, ...) {
-    x <- x$.crunchDataset
-    default.stringsAsFactors <- function () FALSE
-    limit <- min(c(10000, getOption("crunch.data.frame.limit")))
-    if (nrow(x) * ncol(x) > limit) {
-        ## TODO: switch to downloading CSV and reading that?
-        halt("Dataset too large to coerce to data.frame. ",
-            "Consider subsetting it first")
-    }
-    out <- lapply(x, as.vector)
-    names(out) <- names(x)
-    return(structure(out, class="data.frame", row.names=c(NA, -nrow(x))))
+  x <- x$.crunchDataset
+  default.stringsAsFactors <- function () FALSE
+  limit <- min(c(10000, getOption("crunch.data.frame.limit")))
+  if (nrow(x) * ncol(x) > limit) {
+    ## TODO: switch to downloading CSV and reading that?
+    halt("Dataset too large to coerce to data.frame. ",
+         "Consider subsetting it first")
+  }
+  out <- lapply(x, as.vector)
+  names(out) <- names(x)
+  return(structure(out, class="data.frame", row.names=c(NA, -nrow(x))))
 }
+
+
+#' as.data.frame method for VariableCatalog
+#'
+#' This method allows you to import the VariableCatalog into your R sessions in 
+#' order to facilliate interactive use. Modifying the dataframe produced by this
+#' method will not update the Crunch web app, for information about modifying 
+#' variables in the app see: \code{vingette("variables",
+#' package = "crunch")}  
+#' 
+#' @param x
+#' A VariableCatalog produced by \code{variables(ds)}.
+#' @param ... Additional arguments passed to \code{data.frame}
+#' @param row.names part of as.data.frame signature. Ignored.
+#' @param optional part of as.data.frame signature. Ignored.
+#' @param fields 
+#' A character vector of the fields from the variable catalog which you would like 
+#' included in the datafame. To include all fields or see which fields are available
+#' set fields to "all". 
+#' @return A data frame
+#' @examples 
+#' \dontrun{
+#' ds <- loadDataset("iris")
+#' vars <- variables(ds)
+#' var_df <- as.data.frame(vars, fields = "all")
+#' }
+#' 
+#' @rdname VariableCatalog-to-Data-Frame
+#' @export
+as.data.frame.VariableCatalog <- function(x, 
+                                          row.names = NULL, 
+                                          optional = FALSE, 
+                                          ...,
+                                          fields = c("alias", "name", "type")) {
+  out <- catalogToDataFrame(x, rownames = NA)
```

Comment:
Why not just pass "fields" to `catalogToDataFrame` (the arg there is called "keys" but you could rename it to match)?